openfasoc/generators/glayout/glayout/flow/blocks/elementary/cascode_common_source/cascode_common_source.py

```diff
@@ -0,0 +1,407 @@
+
+	# top_level.add_ports(srcM1bulk.get_ports_list(), prefix="VSS")
+	# top_level.add_ports(srcM2bulk.get_ports_list(), prefix="VSS")
+
+	### Adding Port Pins for the cascode common source
+	## VIN input pin placement
+	pin_size = (0.35,0.35)
+	vin_pin = top_level << rectangle(size=pin_size, layer=pdk.get_glayer("met3"), centered=True)
+	vin_pin.movey(top_level.ymin+0.5*evaluate_bbox(vin_pin)[1])
+	vin_pin.movex(-evaluate_bbox(vin_pin)[0]-0.5*evaluate_bbox(M1_ref)[0])
+	## VIN input pin routing from M1 Gate
+	top_level << smart_route(pdk, top_level.ports["M1_gate_W"], vin_pin.ports["e4"], viaoffset=False)
+	## VIN input pin add label for port
+	top_level.add_ports(vin_pin.get_ports_list(), prefix="VIN")
+
+	## VBIAS input pin
+	vbias_pin = top_level << rectangle(size=pin_size, layer=pdk.get_glayer("met3"), centered=True)
+	vbias_pin.movey(evaluate_bbox(M2_ref)[1])
+	vbias_pin.movex(-evaluate_bbox(vin_pin)[0]-0.5*evaluate_bbox(M2_ref)[0])
+	## VBIAS input pin routing from M1 Gate
+	top_level << smart_route(pdk, top_level.ports["M2_gate_W"], vbias_pin.ports["e4"], viaoffset=False)
+	## VBIAS input pin add label for port
+	top_level.add_ports(vbias_pin.get_ports_list(), prefix="VBIAS")
+
+
+	## IOUT pin
+	iout_pin = top_level << rectangle(size=pin_size, layer=pdk.get_glayer("met3"), centered=True)
+	iout_pin.movey(top_level.ymax-0.5*evaluate_bbox(iout_pin)[1])
+	iout_pin.movex(evaluate_bbox(vin_pin)[0]+0.5*evaluate_bbox(M2_ref)[0])
+	## IOUT output pin routing from M1 Gate
+	top_level << smart_route(pdk, top_level.ports["M2_drain_W"], iout_pin.ports["e1"], viaoffset=False)
+	## IOUT output pin add label for port
+	top_level.add_ports(iout_pin.get_ports_list(), prefix="IOUT")
+
+	## VSS pin
+	vss_pin = top_level << rectangle(size=pin_size, layer=pdk.get_glayer("met3"), centered=True)
+	vss_pin.movey(top_level.ymin+0.5*evaluate_bbox(vss_pin)[1])
+	vss_pin.movex(evaluate_bbox(vin_pin)[0]+0.5*evaluate_bbox(M1_ref)[0])
+	## VSS pin routing from M1 Gate
+	top_level << smart_route(pdk, top_level.ports["M1_source_W"], vss_pin.ports["e1"], viaoffset=False)
+	## VSS ad label for port
+	top_level.add_ports(vss_pin.get_ports_list(), prefix="VSS")
+
+	## Bulk connections of both FETs to VSS
+	# top_level << smart_route(pdk, top_level.ports["M1_well_E"],  top_level.ports["M2_well_E"], viaoffset=False)
+
+	
+	top_level.info['netlist'] = cascode_common_source_netlist(
+		pdk, 
+  		m1_width=3, #kwargs.get('width', 21), 
+		m2_width=3, #kwargs.get('width', 21), 
+		m1_length=pdk.get_grule('poly')['min_width'], #kwargs.get('length',14), 
+		m2_length=pdk.get_grule('poly')['min_width'], #kwargs.get('length',14),
+		multipliers=1, 
+    	n_or_p_fet=device,
+		subckt_only=True,
+		m1_fingers = m1_fingers,
+		m2_fingers = m2_fingers,
+		m1_multipliers = m1_multipliers,
+		m2_multipliers = m2_multipliers
+	)
+
+	
+
+	generated_netlist_for_lvs = top_level.info['netlist'].generate_netlist()
+	print(f"Type of generated netlist is :", generated_netlist_for_lvs)
```

Comment:
It would be good if this is done outside the function. This function returns a component, the extra code may cause unintended side effects.